contracts/contracts/lib/hubble-contracts/contracts/libs/BLS.sol

```diff
@@ -39,7 +39,33 @@ library BLS {
     uint256 private constant MASK24 = 0xffffffffffffffffffffffffffffffffffffffffffffffff;
 
     // estimator address
-    address private constant COST_ESTIMATOR_ADDRESS = 0x22E4a5251C1F02de8369Dd6f192033F6CB7531A4;
+    address private constant COST_ESTIMATOR_ADDRESS = 0x6eb8F8d661eFe36daB11147830A1e690249bB830;
+    address private constant COST_ESTIMATOR_ADDRESS_1337 = 0xae0d9c2B7514577D3Ff21e71DC6a9D165cFA6D06;
+
+    /**
+     * This is a temporary workaround while we wait for Safe's singleton factory
+     * deployment signature for chain id 1337:
+     * https://github.com/safe-global/safe-singleton-factory/issues/97
+     * 
+     * In the meantime, this allows us to do fast testing using geth dev mode,
+     * just without getting the same address in this test environment.
+     * 
+     * This adds:
+     * - 1824 L1 gas ($0.05) to the deployment of BLSOpen (114 bytes)
+     * - 71 L2 gas ($0.000012) to signature verification
+     */
+    function getCostEstimator() internal pure returns (address) {
+        uint256 id;
+        assembly {
+            id := chainid()
+        }
+
+        if (id == 1337) {
+            return COST_ESTIMATOR_ADDRESS_1337;
+        }
+
+        return COST_ESTIMATOR_ADDRESS;
+    }
```

Comment:
I hear you, but I think you are overlooking two things:
1. The 1337 address is still hardcoded to an address which can *only* contain the same logic, due to the deterministic deployment. Messing with the chainid does not allow an attacker to insert a different cost estimator.
2. This is not intended to ever deploy to a non-test network.

I wouldn't be happy with modifying the file when using 1337 deployments, because I think that leads to a lot of subtle confusion, as we experienced previously when the committed address here was correct for production but not development.

If you have any contacts at safe, it would be really nice if they just provided the signature we need. It should really only take a few minutes of their time and avoid all this mess.

Reply:
@jacque006 Moot now, Safe came through and I deleted all this code 🎉.